File: scala/src/main/scala/problem1.scala
object problem1{
  def apply(upperBound: Int) : Int = {
    Range(1, upperBound).filter((num:Int) => multipleOf(5)(num) || multipleOf(3)(num)).sum
  } 
}

object multipleOf{
  def apply(divisor: Int)(num: Int): Boolean = num % divisor == 0
}

File: clojure/src/msgehard/projectEuler/problem1.clj
(ns msgehard.projectEuler.problem1)

(defn problem1 [upperBound]
  (let [numbers (range 1 upperBound)
       divisibleBy (fn [divisor num] (= 0 (rem num divisor)))
       divisibleBy3Or5 (fn [num] (or (divisibleBy 3 num ) (divisibleBy 5 num)))
       ]
    (apply + (filter divisibleBy3Or5 numbers))))
